### PR TITLE
fix(embeds): enforce Discord limits — thread names, labels field, CI reply (#136)

### LINKED ISSUE #136: fix(embeds): Discord limit overflows — thread names >100 chars, labels field >1024, CI reply >2000
## Summary
Three unbounded-content paths hit hard Discord API limits (audit C5; Builder #2/#3, Discord #1/#5, Tester edge cases 1-3):

1. **Thread names** (limit 100): `` `PR #${pr.number}: ${pr.title.substring(0, 90)}` `` built at 8 sites (pr.ts:127, 192, 236, 307, 408; issue.ts:108, 154, 205). Any issue ≥ #10 or PR ≥ #10000 with a ≥90-char title → 101 chars → API 400, handler throws, job fails. In handleIssueOpened the embed is sent before startThread, so failure orphans an embed with no DB row.
2. **Issue labels field** (limit 1024): `issue.labels.map(...).join(' ')` unbounded (builders.ts:226-232) — label-heavy issue kills the whole embed send.
3. **CI failure reply** (limit 2000): caps step count at 5 but not name length (builders.ts:159-171).

## Fix
One shared `buildThreadName(kind, number, title)` capping total length at 100; replace all 8 sites. `truncateField(value, 1024)` for the labels join with "+N more". Cap the CI reply at 2000. Tests per the audit edge-case catalog.

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

Closes the three real Discord-limit violations from unbounded GitHub content (audit C5 limit half; flagged by Builder, Discord Specialist, and Tester):

- **Thread names** (limit 100): old code truncated the *title* to 90 chars but ignored the prefix — `Issue #10: ` + 90 = 101 chars → API 400, handler throws, notification job fails. Reachable today by any issue ≥ #10 with a long title. New `buildThreadName()` computes the budget from the full name and replaces the string at **all 8 copy-pasted sites** (5 in pr.ts, 3 in issue.ts).
- **Issue labels field** (limit 1024): unbounded join killed the whole embed send for label-heavy issues. Now capped with a `+N more` tail.
- **CI failure reply** (limit 2000): step *count* was capped, name *length* wasn't — 5 long matrix-job names overflowed. Names truncate at 80 chars + hard cap on the assembled message.

Closes #136

## Test Plan

- [x] TDD: 6 of 8 new tests RED before the fix, GREEN after (`limits.test.ts`)
- [x] Full suite passes, typecheck clean, dist/ rebuilt

### CODE REVIEW

**Comment by blamechris** [comment]
## Code Review: PR #153

### Summary
Closes the three Discord hard-limit overflows: thread names via a shared buildThreadName() (8 sites), labels field cap with +N more, CI failure reply name truncation + hard cap.

### Strengths
- buildThreadName computes the title budget from the full name — the class of bug (prefix ignored) is structurally gone, not patched per-site
- Replacing 8 copy-pasted strings with one helper is the first concrete step toward #145's dedup goal
- The belt-and-suspenders hard cap on the CI reply means no input combination can produce a >2000 send, even future format changes

### Issues Found

#### Critical (Must Fix)
None.

#### Suggestions (Should Consider)
| File | Line | Suggestion | Rationale |
|------|------|------------|-----------|
| builders.ts | formatLabelsField | The 12-char tail reserve is conservative; could compute exactly | Current behavior is correct (under-fills by a few chars at worst); precision not worth the complexity |
| builders.ts | truncateTitle/buildThreadName | substring can split a surrogate pair at the boundary (Tester edge case 16) | Pre-existing in truncateTitle too; worth a small follow-up across all truncators rather than fixing one |

#### Nitpicks (Optional)
- None

### Architecture Notes
buildThreadName lives in builders.ts beside the other limit helpers — consistent placement. issue.test.ts's module mock gained the new export with equivalent behavior.

### Test Coverage
- Typecheck: Pass
- Build: Pass (dist rebuilt + committed)
- Tests: 228/228 (+8 in limits.test.ts, 6 RED→GREEN)

### Verdict
- [x] Approve - Ready to merge

File: src/embeds/builders.ts [hunk 1]
```diff
@@ -547,6 +552,40 @@ function truncateTitle(title: string): string {
...
+ * Build a thread name that fits Discord's 100-char limit. The prefix length
+ * varies with the entity number, so the title budget must be computed from
+ * the full name — truncating the title alone overflows for large numbers.
+ */
+export function buildThreadName(kind: 'PR' | 'Issue', number: number, title: string): string {
+  const prefix = `${kind} #${number}: `;
+  const room = THREAD_NAME_LIMIT - prefix.length;
+  const fitted = title.length > room ? title.substring(0, room - 1) + '…' : title;
+  return prefix + fitted;
+}
```
[the remark this answers is not in the document]
Comment (blamechris) [reply src/embeds/builders.ts:568]: Fixed in 9eff8cf — negative room is guarded with Math.max and the assembled name gets an unconditional substring clamp to THREAD_NAME_LIMIT, so the invariant holds for any inputs.

File: src/embeds/builders.ts [hunk 2]
```diff
@@ -547,6 +552,40 @@ function truncateTitle(title: string): string {
...
+    const reserve = i < labels.length - 1 ? 12 : 0;
+    if (length + piece.length + 1 + reserve > FIELD_VALUE_LIMIT) {
+      parts.push(`+${labels.length - i} more`);
+      break;
+    }
+    parts.push(piece);
+    length += piece.length + 1;
+  }
+  return parts.join(' ');
+}
```
[the remark this answers is not in the document]
Comment (blamechris) [reply src/embeds/builders.ts:591]: Fixed in 9eff8cf — the reserve is now computed from the actual '+N more' tail string for the current index, and the joined result gets a final clamp to FIELD_VALUE_LIMIT making the cap unconditional.
